drift/lib/src/sqlite3/database.dart

```diff
@@ -149,3 +182,37 @@ class _SqliteVersionDelegate extends DynamicVersionDelegate {
     return Future.value();
   }
 }
+
+class _PreparedStatementsCache {
+  final LruMap<String, CommonPreparedStatement> _cache = LruMap(
```

Comment:
I wonder if we can just use a `LinkedHashMap` for this instead of depending on `quiver`? (Where `entries.first` is the LRU and `entries.last` is the MRU)?

When we insert a new statement and the map is full, we'd remove the first key and dispose it. If it's already in the map, we need to remove it and add it again to move it to the back. But I might be overlooking something here that would cause this to break :D

Reply:
That's a nice idea, I'll give it a try. Since we only need a few methods, and not the full Map interface it could be simple enough.

Reply:
I've added a custom LRU based on LinkedHashMap + a small test.